code/EnvironmentChecker.php

```diff
@@ -88,11 +88,21 @@ function canAccess($member = null, $permission = "ADMIN") {
 	function index() {
 		$response = new SS_HTTPResponse;
 		$result = EnvironmentCheckSuite::inst($this->checkSuiteName)->run();
-		
+
+		// output the result as JSON if requested
+		if(
+			$this->getRequest()->getExtension() == 'json'
+			|| strpos($this->getRequest()->getHeader('Accept'), 'application/json') !== false
+		) {
+			$response->setBody($result->toJSON());
+			$response->addHeader('Content-Type', 'application/json');
+			return $response;
```

Comment:
Hmm this will bypass the emailing functionality below... is this what we want?

Cough `defer sendEmail()` oh wait...